src/control/server/mgmt_svc_test.go

```diff
@@ -731,3 +726,77 @@ func TestMgmtSvc_LeaderQuery(t *testing.T) {
 		})
 	}
 }
+
+func TestMgmtSvc_PoolQuery(t *testing.T) {
+	missingSB := newTestMgmtSvc(nil)
+	missingSB.harness.instances[0]._superblock = nil
+
+	for name, tc := range map[string]struct {
+		mgmtSvc *mgmtSvc
+		req     *mgmtpb.PoolQueryReq
+		expResp *mgmtpb.PoolQueryResp
+		expErr  error
+	}{
+		"nil request": {
+			expErr: errors.New("nil request"),
+		},
+		"missing superblock": {
+			mgmtSvc: missingSB,
+			req: &mgmtpb.PoolQueryReq{
+				Uuid: mockUUID,
+			},
+			expErr: errors.New("not an access point"),
+		},
+		"dRPC send fails": {
+			req: &mgmtpb.PoolQueryReq{
+				Uuid: mockUUID,
+			},
+			expErr: errors.New("send failure"),
+		},
+		"garbage req": {
+			req: &mgmtpb.PoolQueryReq{
+				Uuid: "garbage",
+			},
+			expErr: errors.New("unmarshal"),
+		},
+		"successful query": {
+			req: &mgmtpb.PoolQueryReq{
+				Uuid: mockUUID,
+			},
+			expResp: &mgmtpb.PoolQueryResp{
+				Uuid: mockUUID,
+			},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			log, buf := logging.NewTestLogger(t.Name())
+			defer common.ShowBufferOnFailure(t, buf)
+
+			if tc.mgmtSvc == nil {
+				tc.mgmtSvc = newTestMgmtSvc(log)
+			}
+			tc.mgmtSvc.log = log
+
+			if _, err := tc.mgmtSvc.harness.GetMSLeaderInstance(); err == nil {
+				if tc.req != nil && tc.req.Uuid == "garbage" {
+					// dRPC call returns junk in the message body
+					badBytes := makeBadBytes(42)
```

Comment:
Not really crazy about using the request fields to trigger the tests to set up a junk response. I'd rather have some explicit field - maybe even a setupMockDrpc function that's passed as part of the test case.

Reply:
I'll clean this up in #1579 unless this PR needs more love to get through CI.